### PR TITLE
5.0.0 fix

### PR DESCRIPTION
**Pull request by howzat**
In this commit I've rebased the latest changes from the main pusher repo, and then merged the current changes from the active PR branch (optional-id-tokens). The intention is to make the changes applied from both forks explicit.

### CODE REVIEW

File: ci/pipelines/build-and-release.yaml [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+    source:
+      repository: resource/github-status
+
+resources:
+  - name: oauth2_proxy
+    type: git
+    icon: github-circle
+    source:
+      uri: git@github.com:states-of-guernsey/oauth2_proxy.git
+      branch: 4.0.0-fix
```
Comment (sneakybeaky) [inline ci/pipelines/build-and-release.yaml:23]: This will have to be master now I think ?

File: ci/pipelines/build-and-release.yaml [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      private_key: ((github_private_key))
+      ignore_paths:
+        - ci/
+
+  - name: ci
+    type: git
+    icon: github-circle
+    source:
+      uri: git@github.com:states-of-guernsey/oauth2_proxy.git
+      branch: 4.0.0-fix
```
Comment (sneakybeaky) [inline ci/pipelines/build-and-release.yaml:33]: Same - will need to be master ?

File: ci/pipelines/build-and-release.yaml [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  - name: version
+    type: semver
+    icon: tag
+    source:
+      region_name: eu-west-1
+      bucket: ((build_artifacts_bucket.bucket_name))
+      access_key_id: ((build_artifacts_bucket.access_key_id))
+      secret_access_key: ((build_artifacts_bucket.secret_access_key))
+      initial_version: 4.0.0
```
Comment (sneakybeaky) [inline ci/pipelines/build-and-release.yaml:53]: I'd bump this to 5.0.0. This is only ever used if the semver resource isn't already populated, which it is, but just being careful

**Review by sneakybeaky — COMMENTED** [review]
I'd suggest bumping the semver resource to the next major version to be in line, so it becomes 5.x.x on our release. This can be done via the `semver` tab on the concourse pipeline

**Comment by howzat** [comment]
All good points. I'm going to close this PR, talking with @frankoid it would be more sensible to only include those commits that are part of the 5 release (this has everything that is on master). I'll pull those commits over, merge those from my fork - and then make those changes above. Then I'll re-raise the PR

**Pull request closed without merging**